# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/sessionstore/SessionStoreListener.cpp
# Commit: 52ac63af39a4
# Full Hash: 52ac63af39a4de5a2e8ea8d44071a3006f6a1b8d
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2020-04-25 09:49:23
# Regressor Bug: 1580565
# File Overlap Count: 3
# Description:
#   Bug 1580565 - Part 4: Use WindowContext to manage BrowsingContext cached status, r=farre
#   
#   The existing infrastructure which stored cached BrowsingContexts on the
#   BrowsingContextGroup was added before WindowContexts were added, and can cause
#   racing issues with partially discarded trees during process switches.
# ==============================================================================

diff -r 775fcd064844 -r 52ac63af39a4 toolkit/components/sessionstore/SessionStoreListener.cpp
--- a/toolkit/components/sessionstore/SessionStoreListener.cpp	Fri Apr 24 18:33:01 2020 +0000
+++ b/toolkit/components/sessionstore/SessionStoreListener.cpp	Fri Apr 24 18:33:04 2020 +0000
@@ -487,7 +487,7 @@
 
   /* Collect data from all child frame */
   // This is not going to work for fission. Bug 1572084 for tracking it.
-  for (auto& child : aBrowsingContext->GetChildren()) {
+  for (auto& child : aBrowsingContext->Children()) {
     aPositionDescendants[currentIdx] +=
         CollectPositions(child, aPositions, aPositionDescendants);
   }
@@ -576,7 +576,7 @@
 
   /* Collect data from all child frame */
   // This is not going to work for fission. Bug 1572084 for tracking it.
-  for (auto& child : aBrowsingContext->GetChildren()) {
+  for (auto& child : aBrowsingContext->Children()) {
     aInputs[currentIdx].descendants +=
         CollectInputs(child, aInputs, aIdVals, aXPathVals);
   }